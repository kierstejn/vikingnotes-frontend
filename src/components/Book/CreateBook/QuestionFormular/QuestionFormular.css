.QuestionFormular {
    height: 100%;
    width: 100%;
    padding: 20px;
    min-width: 600px;
    max-width: 800px;
}




