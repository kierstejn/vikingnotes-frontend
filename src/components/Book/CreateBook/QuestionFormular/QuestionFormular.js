import React from 'react';
import PropTypes from 'prop-types';

import './QuestionFormular.css';
import Box from '../../../shared/UI/Box/Box';
import TextArea from '../../../shared/UI/TextArea/TextArea';
import Input from '../../../shared/UI/Input/Input';
import ImageUpload from '../ImageUpload/ImageUpload';
import {validate} from "../../../../util/validate/validate";

const questionformular = ({...props}) => {

    let formularStyle = {
        display: 'grid',
        gridTemplateColumns: '1fr 1fr',
        gridTemplateRows: '35px 100px 35px 35px 35px auto',
        gridRowGap: '30px',
    };

    let textAreaStyle = {
        gridColumnStart: 1,
        gridColumnEnd: 3,
        gridRow: 2
    };

    let url = '';
    if(props.question.image !== null) {
        url = URL.createObjectURL(props.question.image);
    }

    let imageStyle = {
        gridRow: 6,
        gridColumn: '2/3',
        width: '60%',
        margin: '0 auto'
    };


    let formular = (
        <div className={"QuestionFormular"} style={formularStyle}>

			<label style={{gridRow: 1, gridColumn: '1/3'}}>
				Kapitel navn (valgfrit):
				<Input
					name={"question"}
					value={props.question.question}
					onChange={props.onQuestionChangedHandler}
					autoComplete={"off"}
					showValidity={props.showValidity}
					valid={validate(props.rules.question, props.question.question)}
				/>
			</label>
            <label style={textAreaStyle}>
                Beskrivelse (valgfrit):
                <TextArea
                    name={"caseText"}
                    value={props.question.caseText}
                    onChange={props.onCaseTextChangeHandler}
                    showValidity={props.showValidity}
                    valid={validate(props.rules.caseText, props.question.caseText)}
                />
            </label>


        </div>
    );

    return(
        <Box>
            {formular}
        </Box>
    )
};

questionformular.propTypes = {
    caseText: PropTypes.string,
    rules: PropTypes.object,
    showValidity: PropTypes.bool,
    onDeleteImageHandler: PropTypes.func,
    onImageSubmittedHandler: PropTypes.func,
    question: PropTypes.object
};

export default questionformular;
