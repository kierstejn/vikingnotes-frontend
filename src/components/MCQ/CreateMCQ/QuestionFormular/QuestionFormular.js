import React from 'react';
import PropTypes from 'prop-types';

import './QuestionFormular.css';
import Box from '../../../shared/UI/Box/Box';
import TextArea from '../../../shared/UI/TextArea/TextArea';
import Input from '../../../shared/UI/Input/Input';
import ImageUpload from '../ImageUpload/ImageUpload';
import {validate} from "../../../../util/validate/validate";

const questionformular = ({...props}) => {

    let formularStyle = {
        display: 'grid',
        gridTemplateColumns: '1fr 1fr',
        gridTemplateRows: '100px 35px 35px 35px 35px auto',
        gridRowGap: '30px',
    };

    let textAreaStyle = {
        gridColumnStart: 1,
        gridColumnEnd: 3,
        gridRow: 1
    };

    let url = '';
    if(props.question.image !== null) {
        url = URL.createObjectURL(props.question.image);
    }

    let imageStyle = {
        gridRow: 6,
        gridColumn: '2/3',
        width: '60%',
        margin: '0 auto'
    };


    let formular = (
        <div className={"QuestionFormular"} style={formularStyle}>
            <label style={textAreaStyle}>
                Case (valgfrit):
                <TextArea
                    name={"caseText"}
                    value={props.question.caseText}
                    onChange={props.onCaseTextChangeHandler}
                    showValidity={props.showValidity}
                    valid={validate(props.rules.caseText, props.question.caseText)}
                />
            </label>
            <label style={{gridRow: 2, gridColumn: '1/3'}}>
                Spørgsmål:
                <Input
                    name={"question"}
                    value={props.question.question}
                    onChange={props.onQuestionChangedHandler}
                    autoComplete={"off"}
                    showValidity={props.showValidity}
                    valid={validate(props.rules.question, props.question.question)}
                />
            </label>

            <label style={{gridRow: 3, gridColumn: '1/3'}}>
                Svar mulighed 1 (rigtigt svar):
                <Input
                    name={"answer"}
                    id={0}
                    onChange={props.onAnswersChangedHandler}
                    value={props.question.answers[0].value}
                    autoComplete={"off"}
                    showValidity={props.showValidity}
                    valid={validate(props.rules.answer, props.question.answers[0].value)}
                />
            </label>

            <label style={{gridRow: 4, gridColumn: '1/3'}}>
                Svar mulighed 2:
                <Input
                    name={"answers"}
                    id={1}
                    onChange={props.onAnswersChangedHandler}
                    value={props.question.answers[1].value}
                    autoComplete={"off"}
                    showValidity={props.showValidity}
                    valid={validate(props.rules.answer, props.question.answers[1].value)}
                />
            </label>

            <label style={{gridRow: 5, gridColumn: '1/3'}}>
                Svar mulighed 3:
                <Input
                    name={"answer"}
                    id={2}
                    onChange={props.onAnswersChangedHandler}
                    value={props.question.answers[2].value}
                    autoComplete={"off"}
                    showValidity={props.showValidity}
                    valid={validate(props.rules.answer, props.question.answers[2].value)}
                />
            </label>
            <ImageUpload
                image={props.question.image}
                onSubmit={props.onImageSubmittedHandler}
                onDelete={props.onDeleteImageHandler}
                style={{gridRow: 6, gridColumn: '1/2', marginTop: '5px', maxHeight: '40px'}}
            />
            {props.question.image ? <img id={"image"} src={url} style={imageStyle} alt={"Preview"}/> : null}

        </div>
    );

    return(
        <Box>
            {formular}
        </Box>
    )
};

questionformular.propTypes = {
    caseText: PropTypes.string,
    rules: PropTypes.object,
    showValidity: PropTypes.bool,
    onDeleteImageHandler: PropTypes.func,
    onImageSubmittedHandler: PropTypes.func,
    question: PropTypes.object
};

export default questionformular;
