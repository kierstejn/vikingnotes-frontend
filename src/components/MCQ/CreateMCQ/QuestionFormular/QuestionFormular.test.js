import React from 'react';

import { shallow } from 'enzyme';

import QuestionFormular from './QuestionFormular';

const questionProp = {
    image: null,
    caseText: '',
    answers: [
        {
            value: ''
        },
        {
            value: ''
        },
        {
            value: ''
        },
    ]
};

const rulesProp = {
    title: {
        required: true
    },
    semester: {
        required: true
    },
    subject: {
        required: true
    },
    description: {
        required: true
    },
    caseText: {
        required: false
    },
    question: {
        required: true
    },
    answer: {
        required: true
    },
    questions: {
        minLenght: 1
    }
};



describe("QuestionFormular - Test", () => {

    it("Renders without crashing", () => {
        let wrapper = shallow(<QuestionFormular question={questionProp} rules={rulesProp}/>);
        expect(wrapper).toMatchSnapshot();
    })

    it("Expect no image when image is null", () => {
        let wrapper = shallow(<QuestionFormular question={questionProp} rules={rulesProp}/>);
        expect(wrapper.find('[id="image"]').exists()).toBe(false);
    })

    it("Expect image when image is not null", () => {
        global.URL.createObjectURL = jest.fn();
        const question = {...questionProp, image: new File(["mock"], "image", { type: 'image/jpeg' })};
        let wrapper = shallow(<QuestionFormular question={question} rules={rulesProp}/>);
        expect(wrapper.find('[id="image"]').exists()).toBe(true);
    })

});
