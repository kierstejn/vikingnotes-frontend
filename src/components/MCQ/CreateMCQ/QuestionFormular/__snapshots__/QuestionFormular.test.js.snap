// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionFormular - Test Renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <questionformular
    question={
      Object {
        "answers": Array [
          Object {
            "value": "",
          },
          Object {
            "value": "",
          },
          Object {
            "value": "",
          },
        ],
        "caseText": "",
        "image": null,
      }
    }
    rules={
      Object {
        "answer": Object {
          "required": true,
        },
        "caseText": Object {
          "required": false,
        },
        "description": Object {
          "required": true,
        },
        "question": Object {
          "required": true,
        },
        "questions": Object {
          "minLenght": 1,
        },
        "semester": Object {
          "required": true,
        },
        "subject": Object {
          "required": true,
        },
        "title": Object {
          "required": true,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="QuestionFormular"
        style={
          Object {
            "display": "grid",
            "gridRowGap": "30px",
            "gridTemplateColumns": "1fr 1fr",
            "gridTemplateRows": "100px 35px 35px 35px 35px auto",
          }
        }
      >
        <label
          style={
            Object {
              "gridColumnEnd": 3,
              "gridColumnStart": 1,
              "gridRow": 1,
            }
          }
        >
          Case (valgfrit):
          <textArea
            name="caseText"
            rules={
              Object {
                "required": false,
              }
            }
            value=""
          />
        </label>
        <label
          style={
            Object {
              "gridColumn": "1/3",
              "gridRow": 2,
            }
          }
        >
          Spørgsmål:
          <input
            autoComplete="off"
            name="question"
            rules={
              Object {
                "required": true,
              }
            }
          />
        </label>
        <label
          style={
            Object {
              "gridColumn": "1/3",
              "gridRow": 3,
            }
          }
        >
          Svar mulighed 1 (rigtigt svar):
          <input
            autoComplete="off"
            id={0}
            name="answer"
            rules={
              Object {
                "required": true,
              }
            }
            value=""
          />
        </label>
        <label
          style={
            Object {
              "gridColumn": "1/3",
              "gridRow": 4,
            }
          }
        >
          Svar mulighed 2:
          <input
            autoComplete="off"
            id={1}
            name="answers"
            rules={
              Object {
                "required": true,
              }
            }
            value=""
          />
        </label>
        <label
          style={
            Object {
              "gridColumn": "1/3",
              "gridRow": 5,
            }
          }
        >
          Svar mulighed 3:
          <input
            autoComplete="off"
            id={2}
            name="answer"
            rules={
              Object {
                "required": true,
              }
            }
            value=""
          />
        </label>
        <ImageUpload
          image={null}
          style={
            Object {
              "gridColumn": "1/2",
              "gridRow": 6,
              "marginTop": "5px",
              "maxHeight": "40px",
            }
          }
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            style={
              Object {
                "gridColumnEnd": 3,
                "gridColumnStart": 1,
                "gridRow": 1,
              }
            }
          >
            Case (valgfrit):
            <textArea
              name="caseText"
              rules={
                Object {
                  "required": false,
                }
              }
              value=""
            />
          </label>,
          <label
            style={
              Object {
                "gridColumn": "1/3",
                "gridRow": 2,
              }
            }
          >
            Spørgsmål:
            <input
              autoComplete="off"
              name="question"
              rules={
                Object {
                  "required": true,
                }
              }
            />
          </label>,
          <label
            style={
              Object {
                "gridColumn": "1/3",
                "gridRow": 3,
              }
            }
          >
            Svar mulighed 1 (rigtigt svar):
            <input
              autoComplete="off"
              id={0}
              name="answer"
              rules={
                Object {
                  "required": true,
                }
              }
              value=""
            />
          </label>,
          <label
            style={
              Object {
                "gridColumn": "1/3",
                "gridRow": 4,
              }
            }
          >
            Svar mulighed 2:
            <input
              autoComplete="off"
              id={1}
              name="answers"
              rules={
                Object {
                  "required": true,
                }
              }
              value=""
            />
          </label>,
          <label
            style={
              Object {
                "gridColumn": "1/3",
                "gridRow": 5,
              }
            }
          >
            Svar mulighed 3:
            <input
              autoComplete="off"
              id={2}
              name="answer"
              rules={
                Object {
                  "required": true,
                }
              }
              value=""
            />
          </label>,
          <ImageUpload
            image={null}
            style={
              Object {
                "gridColumn": "1/2",
                "gridRow": 6,
                "marginTop": "5px",
                "maxHeight": "40px",
              }
            }
          />,
          null,
        ],
        "className": "QuestionFormular",
        "style": Object {
          "display": "grid",
          "gridRowGap": "30px",
          "gridTemplateColumns": "1fr 1fr",
          "gridTemplateRows": "100px 35px 35px 35px 35px auto",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Case (valgfrit):",
              <textArea
                name="caseText"
                rules={
                  Object {
                    "required": false,
                  }
                }
                value=""
              />,
            ],
            "style": Object {
              "gridColumnEnd": 3,
              "gridColumnStart": 1,
              "gridRow": 1,
            },
          },
          "ref": null,
          "rendered": Array [
            "Case (valgfrit):",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "changed": undefined,
                "name": "caseText",
                "rules": Object {
                  "required": false,
                },
                "showValidity": undefined,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Spørgsmål:",
              <input
                autoComplete="off"
                name="question"
                rules={
                  Object {
                    "required": true,
                  }
                }
              />,
            ],
            "style": Object {
              "gridColumn": "1/3",
              "gridRow": 2,
            },
          },
          "ref": null,
          "rendered": Array [
            "Spørgsmål:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoComplete": "off",
                "changed": undefined,
                "name": "question",
                "rules": Object {
                  "required": true,
                },
                "showValidity": undefined,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Svar mulighed 1 (rigtigt svar):",
              <input
                autoComplete="off"
                id={0}
                name="answer"
                rules={
                  Object {
                    "required": true,
                  }
                }
                value=""
              />,
            ],
            "style": Object {
              "gridColumn": "1/3",
              "gridRow": 3,
            },
          },
          "ref": null,
          "rendered": Array [
            "Svar mulighed 1 (rigtigt svar):",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoComplete": "off",
                "changed": undefined,
                "id": 0,
                "name": "answer",
                "rules": Object {
                  "required": true,
                },
                "showValidity": undefined,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Svar mulighed 2:",
              <input
                autoComplete="off"
                id={1}
                name="answers"
                rules={
                  Object {
                    "required": true,
                  }
                }
                value=""
              />,
            ],
            "style": Object {
              "gridColumn": "1/3",
              "gridRow": 4,
            },
          },
          "ref": null,
          "rendered": Array [
            "Svar mulighed 2:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoComplete": "off",
                "changed": undefined,
                "id": 1,
                "name": "answers",
                "rules": Object {
                  "required": true,
                },
                "showValidity": undefined,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Svar mulighed 3:",
              <input
                autoComplete="off"
                id={2}
                name="answer"
                rules={
                  Object {
                    "required": true,
                  }
                }
                value=""
              />,
            ],
            "style": Object {
              "gridColumn": "1/3",
              "gridRow": 5,
            },
          },
          "ref": null,
          "rendered": Array [
            "Svar mulighed 3:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoComplete": "off",
                "changed": undefined,
                "id": 2,
                "name": "answer",
                "rules": Object {
                  "required": true,
                },
                "showValidity": undefined,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "image": null,
            "onDelete": undefined,
            "onSubmit": undefined,
            "style": Object {
              "gridColumn": "1/2",
              "gridRow": 6,
              "marginTop": "5px",
              "maxHeight": "40px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="QuestionFormular"
          style={
            Object {
              "display": "grid",
              "gridRowGap": "30px",
              "gridTemplateColumns": "1fr 1fr",
              "gridTemplateRows": "100px 35px 35px 35px 35px auto",
            }
          }
        >
          <label
            style={
              Object {
                "gridColumnEnd": 3,
                "gridColumnStart": 1,
                "gridRow": 1,
              }
            }
          >
            Case (valgfrit):
            <textArea
              name="caseText"
              rules={
                Object {
                  "required": false,
                }
              }
              value=""
            />
          </label>
          <label
            style={
              Object {
                "gridColumn": "1/3",
                "gridRow": 2,
              }
            }
          >
            Spørgsmål:
            <input
              autoComplete="off"
              name="question"
              rules={
                Object {
                  "required": true,
                }
              }
            />
          </label>
          <label
            style={
              Object {
                "gridColumn": "1/3",
                "gridRow": 3,
              }
            }
          >
            Svar mulighed 1 (rigtigt svar):
            <input
              autoComplete="off"
              id={0}
              name="answer"
              rules={
                Object {
                  "required": true,
                }
              }
              value=""
            />
          </label>
          <label
            style={
              Object {
                "gridColumn": "1/3",
                "gridRow": 4,
              }
            }
          >
            Svar mulighed 2:
            <input
              autoComplete="off"
              id={1}
              name="answers"
              rules={
                Object {
                  "required": true,
                }
              }
              value=""
            />
          </label>
          <label
            style={
              Object {
                "gridColumn": "1/3",
                "gridRow": 5,
              }
            }
          >
            Svar mulighed 3:
            <input
              autoComplete="off"
              id={2}
              name="answer"
              rules={
                Object {
                  "required": true,
                }
              }
              value=""
            />
          </label>
          <ImageUpload
            image={null}
            style={
              Object {
                "gridColumn": "1/2",
                "gridRow": 6,
                "marginTop": "5px",
                "maxHeight": "40px",
              }
            }
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              style={
                Object {
                  "gridColumnEnd": 3,
                  "gridColumnStart": 1,
                  "gridRow": 1,
                }
              }
            >
              Case (valgfrit):
              <textArea
                name="caseText"
                rules={
                  Object {
                    "required": false,
                  }
                }
                value=""
              />
            </label>,
            <label
              style={
                Object {
                  "gridColumn": "1/3",
                  "gridRow": 2,
                }
              }
            >
              Spørgsmål:
              <input
                autoComplete="off"
                name="question"
                rules={
                  Object {
                    "required": true,
                  }
                }
              />
            </label>,
            <label
              style={
                Object {
                  "gridColumn": "1/3",
                  "gridRow": 3,
                }
              }
            >
              Svar mulighed 1 (rigtigt svar):
              <input
                autoComplete="off"
                id={0}
                name="answer"
                rules={
                  Object {
                    "required": true,
                  }
                }
                value=""
              />
            </label>,
            <label
              style={
                Object {
                  "gridColumn": "1/3",
                  "gridRow": 4,
                }
              }
            >
              Svar mulighed 2:
              <input
                autoComplete="off"
                id={1}
                name="answers"
                rules={
                  Object {
                    "required": true,
                  }
                }
                value=""
              />
            </label>,
            <label
              style={
                Object {
                  "gridColumn": "1/3",
                  "gridRow": 5,
                }
              }
            >
              Svar mulighed 3:
              <input
                autoComplete="off"
                id={2}
                name="answer"
                rules={
                  Object {
                    "required": true,
                  }
                }
                value=""
              />
            </label>,
            <ImageUpload
              image={null}
              style={
                Object {
                  "gridColumn": "1/2",
                  "gridRow": 6,
                  "marginTop": "5px",
                  "maxHeight": "40px",
                }
              }
            />,
            null,
          ],
          "className": "QuestionFormular",
          "style": Object {
            "display": "grid",
            "gridRowGap": "30px",
            "gridTemplateColumns": "1fr 1fr",
            "gridTemplateRows": "100px 35px 35px 35px 35px auto",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Case (valgfrit):",
                <textArea
                  name="caseText"
                  rules={
                    Object {
                      "required": false,
                    }
                  }
                  value=""
                />,
              ],
              "style": Object {
                "gridColumnEnd": 3,
                "gridColumnStart": 1,
                "gridRow": 1,
              },
            },
            "ref": null,
            "rendered": Array [
              "Case (valgfrit):",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "changed": undefined,
                  "name": "caseText",
                  "rules": Object {
                    "required": false,
                  },
                  "showValidity": undefined,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Spørgsmål:",
                <input
                  autoComplete="off"
                  name="question"
                  rules={
                    Object {
                      "required": true,
                    }
                  }
                />,
              ],
              "style": Object {
                "gridColumn": "1/3",
                "gridRow": 2,
              },
            },
            "ref": null,
            "rendered": Array [
              "Spørgsmål:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "autoComplete": "off",
                  "changed": undefined,
                  "name": "question",
                  "rules": Object {
                    "required": true,
                  },
                  "showValidity": undefined,
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Svar mulighed 1 (rigtigt svar):",
                <input
                  autoComplete="off"
                  id={0}
                  name="answer"
                  rules={
                    Object {
                      "required": true,
                    }
                  }
                  value=""
                />,
              ],
              "style": Object {
                "gridColumn": "1/3",
                "gridRow": 3,
              },
            },
            "ref": null,
            "rendered": Array [
              "Svar mulighed 1 (rigtigt svar):",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "autoComplete": "off",
                  "changed": undefined,
                  "id": 0,
                  "name": "answer",
                  "rules": Object {
                    "required": true,
                  },
                  "showValidity": undefined,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Svar mulighed 2:",
                <input
                  autoComplete="off"
                  id={1}
                  name="answers"
                  rules={
                    Object {
                      "required": true,
                    }
                  }
                  value=""
                />,
              ],
              "style": Object {
                "gridColumn": "1/3",
                "gridRow": 4,
              },
            },
            "ref": null,
            "rendered": Array [
              "Svar mulighed 2:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "autoComplete": "off",
                  "changed": undefined,
                  "id": 1,
                  "name": "answers",
                  "rules": Object {
                    "required": true,
                  },
                  "showValidity": undefined,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Svar mulighed 3:",
                <input
                  autoComplete="off"
                  id={2}
                  name="answer"
                  rules={
                    Object {
                      "required": true,
                    }
                  }
                  value=""
                />,
              ],
              "style": Object {
                "gridColumn": "1/3",
                "gridRow": 5,
              },
            },
            "ref": null,
            "rendered": Array [
              "Svar mulighed 3:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "autoComplete": "off",
                  "changed": undefined,
                  "id": 2,
                  "name": "answer",
                  "rules": Object {
                    "required": true,
                  },
                  "showValidity": undefined,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "image": null,
              "onDelete": undefined,
              "onSubmit": undefined,
              "style": Object {
                "gridColumn": "1/2",
                "gridRow": 6,
                "marginTop": "5px",
                "maxHeight": "40px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
